Quaternion/Quaternion.py

```diff
@@ -275,38 +326,35 @@ def _quat2equatorial(self):
         """
         Determine Right Ascension, Declination, and Roll for the object quaternion
 
-        :returns: RA, Dec, Roll
+        :returns: N x (RA, Dec, Roll)
         :rtype: numpy array [ra,dec,roll]
         """
 
-        q = self.q
-        q2 = self.q**2
+        q = np.atleast_2d(self.q)
```

Comment:
Yes, but note that nothing is being allocated, because atleast_2d returns a view.

It really depends on what you want. You could prefer to encapsulate all uses of self._q inside _set_q and _get_q, and use self.q everywhere else (the proper way of lazy initialization, IMO). Or you could want to optimize by removing the intermediate calls.

The current implementation has three properties (q, equatorial, transform), with corresponding lazy data members (_q, _equatorial, _transform). The lazy data members are only used in the constructor and the _get/set_ methods, so the get/set methods are tightly bound.

Using self._q here would break the pattern. This would be the one case where a lazy data member is called out of the get/setters. This function (_quat2equatorial) is only called from _get_equatorial, right after setting self._q anyway, so it happens to work. I would argue that it is this statement inside _get_equatorial that is redundant:
self._q = self._transform2quat()

Reply:
Fair enough.  What caught my attention is the undo-ing of `atleast_2d` in `_get_q` (in the scalar case) and then immediately re-doing it here.  But I agree what you have is more robust and consistent.